pkg/controller/plan/validation.go

```diff
@@ -626,3 +632,36 @@ func (r *Reconciler) validateHooks(plan *api.Plan) (err error) {
 
 	return
 }
+
+func (r *Reconciler) validateVddkImage(plan *api.Plan) (err error) {
+	vddkNotConfigured := libcnd.Condition{
+		Type:     VDDKNotConfigured,
+		Status:   True,
+		Reason:   NotSet,
+		Category: Critical,
+		Message:  "VDDK image is necessary for this type of migration",
+	}
+
+	source := plan.Referenced.Provider.Source
+	if source == nil {
+		return nil
+	}
+	destination := plan.Referenced.Provider.Destination
+	if destination == nil {
+		return nil
```

Comment:
I see that we do it this way also in `validateVM` - not sure I understand the reason for this but if we get here and have a plan that is not set with a source/destination provider then I guess they were removed?

Reply:
I guess either removed or not yet created (e.g. from command line). From what I saw we do the check when validating plan and host, but not for network and storage maps.